src/dotnet/APIView/APIViewWeb/Managers/CodeFileManager.cs

```diff
@@ -141,7 +143,7 @@ public async Task<CodeFile> CreateCodeFileAsync(
             Stream fileStream = null,
             string language = null)
         {
-            var languageService = _languageServices.FirstOrDefault(s => (language != null ? s.Name == language : s.IsSupportedFile(originalName)));
+            var languageService = LanguageServiceHelpers.GetLanguageService(language, _languageServices) ?? _languageServices.FirstOrDefault(s => s.IsSupportedFile(originalName));
```

Comment:
When would we ever not find a languageService by language name but have an originalName that does resolve to a service?

Reply:
https://github.com/chidozieononiwu/azure-sdk-tools/blob/e6877552036494f0c22f45dff193c58e2c82d775/src/dotnet/APIView/APIViewWeb/Controllers/AutoReviewController.cs#L141

CreateApiReview does not send the language. But we should definitely be sending the language like you suggested.

Reply:
I created an issue by the way https://github.com/Azure/azure-sdk-tools/issues/10147